### PR TITLE
Bump bazel rules_go version

### PR DESCRIPTION
**Pull request by cblecker**
**What this PR does / why we need it**:
Bump bazel rules_go version. This will allow us to use go1.9.2.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by ixdy** [comment]
We still need https://github.com/kubernetes/repo-infra/pull/50 to be merged. :\

**Comment by cblecker** [comment]
Ohhh right. I thought it already was :(

File: build/root/WORKSPACE [hunk 1]
```diff
@@ -1,15 +1,15 @@
 http_archive(
     name = "io_bazel_rules_go",
-    sha256 = "441e560e947d8011f064bd7348d86940d6b6131ae7d7c4425a538e8d9f884274",
-    strip_prefix = "rules_go-c72631a220406c4fae276861ee286aaec82c5af2",
-    urls = ["https://github.com/bazelbuild/rules_go/archive/c72631a220406c4fae276861ee286aaec82c5af2.tar.gz"],
+    sha256 = "7e748c5fa17248eff598081bcb711172350335e6c4baefafb828fac3a6740f4f",
+    strip_prefix = "rules_go-6d900bc95ae678bec5c91031b8e987957d2a7f93",
+    urls = ["https://github.com/bazelbuild/rules_go/archive/6d900bc95ae678bec5c91031b8e987957d2a7f93.tar.gz"],
```
Comment (ixdy) [inline build/root/WORKSPACE]: what made you choose this version? I wonder if we could instead bump to f4bebf54dca2ad198b0311fd772c8100149cd647 (for go 1.9.2) while some of the performance issues are being worked out.

**Comment by ixdy** [comment]
kubernetes/repo-infra#50 has been merged.

An added complication is that you'll need to change `features = ["-race"]` to `race = "off"` in the two `BUILD` files where that's used, but some recent commits of `rules_go` broke (and then fixed) that.

**Comment by ixdy** [comment]
I'm a little worried we might be hitting https://github.com/bazelbuild/rules_go/issues/1023 still. I guess we can see what Prow does...

**Comment by ixdy** [comment]
In particular, I am worried about `make bazel-test`, which seems to be stuck in the analysis phase.

**Comment by cblecker** [comment]
Urgh. I actually just looked at my test PR, and it never reported back on the bazel-test either (was just looking at the build). Perhaps dialing the rules_go back to https://github.com/bazelbuild/rules_go/commit/f4bebf54dca2ad198b0311fd772c8100149cd647 is a safer bet.

**Comment by ixdy** [comment]
Of course, `rules_go` at that commit doesn't have `go/private:mode.bzl`, so my `go_genrule` fix doesn't work. I think I know how to fix it, though it's frustrating...

**Comment by ixdy** [comment]
I think bazelbuild/rules_go@f4bebf5 might also be broken...

**Comment by ixdy** [comment]
/close

`rules_go` is pretty broken right now, but I think I've found a better way to bump us to go1.9.2.

**Pull request closed without merging**